components/basecomponents/Exercise/exerciseStore.ts

```diff
@@ -0,0 +1,493 @@
+import { fetchRawDictionary, DictionaryDataObject, Phrase, CategoryDataObject } from 'utils/getDataUtils';
+import { getCountryVariant, Language } from 'utils/locales';
+import { AudioPlayer } from 'utils/AudioPlayer';
+import { create } from 'zustand';
+import * as R from 'ramda';
```

Comment:
Is it necessary to use Ramda here? Functional programming is cool and all, but it's a pretty advanced approach to doing things and the syntax and functionality will be unfamiliar to most people. Plus, it's another library newcomers would need to understand to be able to contribute to the code base. In the interest of keeping the code base [beginner friendly](https://github.com/cesko-digital/movapp#development-guidelines), consider removing this dependency a implementing the functionality with simple TypeScript. Can be in a separate PR not to hold up the release.

Reply:
It was recommended for updating nested states in Zustand docs, and I know this lib.

Reply:
Ramda removed.